gateway/src/main/java/io/zeebe/gateway/EndpointManager.java

```diff
@@ -320,6 +326,11 @@ public void updateJobRetries(
       return;
     }
 
+    final ServerCallStreamObserver<GrpcResponseT> serverObserver =
+        (ServerCallStreamObserver<GrpcResponseT>) streamObserver;
+    serverObserver.setOnCancelHandler(
+        () -> Loggers.GATEWAY_LOGGER.trace("gRPC {} request cancelled", grpcRequest.getClass()));
+
     brokerClient.sendRequest(
```

Comment:
Could we somehow reuse the method `sendRequest`, i dont like this duplication.

Reply:
:+1: